yelp_bytes.py

```diff
@@ -26,22 +37,24 @@ def from_bytes(obj, encoding='internet', errors='strict'):
 
     This function always returns unicode.
     """
-    if isinstance(obj, unicode):
+    mro = type(obj).mro()
+    if unicode in mro:
         return obj
+    elif bytes in mro:
+        return obj.decode(encoding, errors)
+    elif hasattr(obj, '__bytes__'):
+        return obj.__bytes__().decode(encoding, errors)
```

Comment:
Previously, we had the invariant that `from_bytes(x) == to_bytes(x).decode('internet')`.
Invariants are a good smell, and i'm loath to break it.

If we can factor that line out, and keep that invariant, i'm for it.
I'll add a test asserting the invariant.